--- original_gstnvinfer.cpp     2023-06-02 06:13:15.296537620 +0000
+++ gstnvinfer.cpp      2023-06-02 06:12:01.481223153 +0000
@@ -1933,6 +1933,15 @@
   }
   std::unordered_map<guint, TensorInputBatch> tensormeta_map;

+  gboolean skip_batch;
+
+  /* Process batch only when interval_counter is 0. */
+  skip_batch = (nvinfer->interval_counter++ % (nvinfer->interval + 1) > 0);
+
+  if (skip_batch) {
+    return GST_FLOW_OK;
+  }
+
   for (NvDsMetaList * l_user = batch_meta->batch_user_meta_list; l_user != NULL;
       l_user = l_user->next) {